### PR TITLE
Remove Deprecated Shared Secret Package

### PR DESCRIPTION
**Pull request by MartinSchoeler**
@RocketChat/core

### CODE REVIEW

File: packages/rocketchat-sharedsecret/sharedsecret.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/* globals HandleSlashCommand */
+const SharedSecret = [];
+
+if (Meteor.isServer) {
+	const EncryptMessage = (() => {
```
Comment (ggazzo) [inline packages/rocketchat-sharedsecret/sharedsecret.js]: If you want to "make a class", I think that you remove/forgot one `()` here 
```javascript
const Class = (()=>{return function(){}})()
```
nowadays you can just use `class EncryptMessage {...}`

but I think that in all situations could be simple functions.
like:
```javascript
function HandleSlashCommand(command, params, item) {...}
```

File: packages/rocketchat-sharedsecret/sharedsecret.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
+const SharedSecret = [];
+
+if (Meteor.isServer) {
+	const EncryptMessage = ((message) => {
```
Comment (rodrigok) [inline packages/rocketchat-sharedsecret/sharedsecret.js]: Lets use named function for theses cases (high level functions / were we used classes before) to help the stack trace.

`function EncryptMessage(message) {`

File: packages/rocketchat-sharedsecret/sharedsecret.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		if (message) {
+			const currentRoomId = message.rid;
+			if (SharedSecret && SharedSecret[currentUser] && SharedSecret[currentUser][currentRoomId]) {
+				const currentSecret = SharedSecret[currentUser][currentRoomId];
+				const encrypted = CryptoJS.AES.encrypt(message.msg, currentSecret);
+				message.msg = encrypted.toString();
+				if (message.urls) {
+					const messageUrls = message.urls;
+					messageUrls.forEach((i) => {
+						const urls = messageUrls[i];
```
Comment (rodrigok) [inline packages/rocketchat-sharedsecret/sharedsecret.js]: The `i` is the `urls` already. Just rename the `i` to `urls` and remove the `const urls = messageUrls[i];`

File: packages/rocketchat-sharedsecret/sharedsecret.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+						urls.url = CryptoJS.AES.encrypt(urls.url, currentSecret).toString();
+					});
+				}
+				message.encrypted = true;
+			}
+		}
+		return message;
+	})();
+
+	const HandleSlashCommand = ((command, params, item) => {
```
Comment (rodrigok) [inline packages/rocketchat-sharedsecret/sharedsecret.js]: Same thing here

File: packages/rocketchat-sharedsecret/sharedsecret.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+				SharedSecret[currentUser][currentRoomId] = secret;
+			}
+		}
+	})();
+	RocketChat.callbacks.add('beforeSaveMessage', EncryptMessage, 9999, 'sharedsecret-encrypt-message');
+	RocketChat.slashCommands.add('setsecretkey', HandleSlashCommand);
+}
+
+if (Meteor.isClient) {
+	const DecryptMessage = ((message) => {
```
Comment (rodrigok) [inline packages/rocketchat-sharedsecret/sharedsecret.js]: Same thing here

File: packages/rocketchat-sharedsecret/sharedsecret.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+					message.html = '~ encrypted message ~';
+				} else {
+					const lockImage = 'images/lock8.png';
+					message.msg = `<img src=${ lockImage } style='width:8px;height:9px;'></img> ${ decrypted }`;
+					message.html = `<img src=${ lockImage } style='width:8px;height:9px;'></img> ${ decrypted }`;
+				}
+				if (message.urls) {
+					const messageUrls = message.urls;
+					messageUrls.forEach((i) => {
+						const urls = messageUrls[i];
```
Comment (rodrigok) [inline packages/rocketchat-sharedsecret/sharedsecret.js]: Same thing here

File: packages/rocketchat-sharedsecret/sharedsecret.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+				}
+			} else {
+				message.msg = '~ encrypted message ~';
+				message.html = '~ encrypted message ~';
+			}
+		}
+		return message;
+	})();
+
+	const HandleSlashCommand = ((command, params, item) => {
```
Comment (rodrigok) [inline packages/rocketchat-sharedsecret/sharedsecret.js]: Same thing here

**Comment by Kiran-Rao** [comment]
This may be a stupid question, but I don't see this package used anywhere. 

It's not within `.meteor/packages`

**Comment by rodrigok** [comment]
@Kiran-Rao You are right.

I'll double check that

**Comment by rodrigok** [comment]
@Kiran-Rao we decided to remove the package, it was there as an optional package.

@MartinSchoeler Can you delete the files and add a note to HISTORY.md about this removal? Like `Removed Deprecated Package rocketchat:sharedsecret`

**Pull request merged**